### DISCUSSION TITLE
Downsample operation for mcap CLI

### DISCUSSION THREAD

**Discussion by saulrh** [foxglove/mcap#1106] [General]
Feature request: A mode to `--filter`, or a new subcommand `mcap downsample`, that downsamples data.
- "Output one record every T interval"
- "Output every Nth record"
- "Output each record with probability P or 1/N"

Would be useful for producing lower-resolution reports for monitoring without having to offload full-res captures.

**Comment by jtbandes** [comment]
Just trying to think through a bit how this would work – would the downsample operations be per-topic, per-channel, or per every message in the file? How would you specify these things at the command line when there are multiple topics in the file? I'm thinking this would be better as a custom python script than an overly complex cli tool.
